File: factories/Abstract/AbstractFactory.ts
interface IButton {}
interface ILabel {}

class BlueButton implements IButton { }
class GreenButton implements IButton { }

class BlueLabel implements ILabel { }
class GreenILabel implements ILabel { }

interface IComponentAbstractFacroty {
    GetButton() : IButton;
    GetLabel() : IButton;
}

class BluComponentAbstractFactory implements IComponentAbstractFacroty{
    GetButton(): IButton {
        return new BlueButton();
    }
    GetLabel(): IButton {
        return new GreenButton();
    }
}

File: factories/Abstract/AbstractFactory.cs
interface IButton {}
interface ILabel {}

class BlueButton : IButton {}
class GreenButton : IButton {}

class BlueLabel : ILabel {}
class GreenLabel : ILabel {}

interface IComponentAbstractFactory{
    IButton GetButton();
    ILabel GetLabel();
}

class BluComponentAbstractFactory : IComponentAbstractFactory {

    public IButton GetButton() => new BlueButton(); 
    public ILabel GetLabel() => new BlueLabel(); 
}

class GreenComponentAbstractFactory : IComponentAbstractFactory {

    public IButton GetButton() => new GreenButton(); 
    public ILabel GetLabel() => new GreenLabel(); 
}